include/arch/arm/cortex_m/scripts/linker.ld

```diff
@@ -415,6 +415,9 @@ SECTIONS
 
     __data_ram_end = .;
 
+#ifdef CONFIG_RAM_FUNCTION
```

Comment:
Is there a particular reason why the RAM code section needs to be at the end, i.e. after kernel SRAM?

I would expect it is simpler if you sneak it in the beginning of SRAM.